### ISSUE TITLE
Union of string sets cannot be converted to dense tensor (raises TypeError)

### ISSUE THREAD

**Issue by ageron** [tensorflow/tensorflow#31009]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
MacOSX 10.13.6
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
N/A
- TensorFlow installed from (source or binary):
binary
- TensorFlow version (use command below):
VERSION=2.0.0-dev20190724
GIT_VERSION=v1.12.1-6931-g2b5ece29d3
- Python version:
3.6.8
- Bazel version (if compiling from source):
N/A
- GCC/Compiler version (if compiling from source):
N/A
- CUDA/cuDNN version:
N/A
- GPU model and memory:
N/A

**Describe the current behavior**
I get an exception when computing the union of two string sets and trying to convert the output to a dense tensor (`TypeError: Cannot convert 0 to EagerTensor of dtype string`).

**Describe the expected behavior**
I should get no exception.

**Code to reproduce the issue**

```python
import tensorflow as tf
a = tf.constant([["a", "b", "c", "", ""], ["d", "e", "", "", ""]])
b = tf.constant([["c", "e", "g", "h", ""], ["d", "", "", "", ""]])
tf.sparse.to_dense(tf.sets.union(a, b))
```

**Other info / logs**
Here's the stacktrace:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-e742e5c0ff6f> in <module>
      2 a = tf.constant([["a", "b", "c", "", ""], ["d", "e", "", "", ""]])
      3 b = tf.constant([["c", "e", "g", "h", ""], ["d", "", "", "", ""]])
----> 4 tf.sparse.to_dense(tf.sets.union(a, b))

~/miniconda3/envs/tf2/lib/python3.6/site-packages/tensorflow_core/python/ops/sparse_ops.py in sparse_tensor_to_dense(sp_input, default_value, validate_indices, name)
   1478       default_value=default_value,
   1479       validate_indices=validate_indices,
-> 1480       name=name)
   1481
   1482

~/miniconda3/envs/tf2/lib/python3.6/site-packages/tensorflow_core/python/ops/gen_sparse_ops.py in sparse_to_dense(sparse_indices, output_shape, sparse_values, default_value, validate_indices, name)
   3158         "SparseToDense", name, _ctx._post_execution_callbacks, sparse_indices,
   3159         output_shape, sparse_values, default_value, "validate_indices",
-> 3160         validate_indices)
   3161       return _result
   3162     except _core._FallbackException:

TypeError: Cannot convert 0 to EagerTensor of dtype string
```

**Comment by ageron** [comment]
Ok, the problem seems to have nothing to do with sets. It's just that the default value in `to_dense()` is 0, which cannot be converted to a string, so the following code works:

```python
import tensorflow as tf
a = tf.constant([["a", "b", "c", "", ""], ["d", "e", "", "", ""]])
b = tf.constant([["c", "e", "g", "h", ""], ["d", "", "", "", ""]])
tf.sparse.to_dense(tf.sets.union(a, b), default_value=b'')
```

I think the default value for string sparse tensors should be the empty string. At the very least, the error message should be clearer (it should say that it's talking about the `default_value`).

**Comment by yongtang** [comment]
@ageron I think this is the same issue as #30750 and will be fixed by PR #30781. The PR has been approved so likely it will be merged soon.

**Comment by ageron** [comment]
Indeed, it is, thanks @yongtang.
